syntax = "proto3";

package memcachekv.proto;

message Operation {
    enum Type {
        GET = 0;
        PUT = 1;
        DEL = 2;
    }
    Type op_type = 1;
    string key = 2;
    string value = 3;
}

message MemcacheKVRequest {
    int32 client_id = 1;
    uint32 req_id = 2;
    Operation op = 3;
}

enum Result {
    OK = 0;
    NOT_FOUND = 1;
}

message MemcacheKVReply {
    int32 node_id = 1;
    int32 client_id = 2;
    uint32 req_id = 3;
    Result result = 4;
    string value = 5;
}

message MemcacheKVMessage {
    oneof msg {
        MemcacheKVRequest request = 1;
        MemcacheKVReply reply = 2;
    }
}
